api/src/main/java/org/infinispan/api/common/AsyncIterator.java

```diff
@@ -0,0 +1,11 @@
+package org.infinispan.api.common;
+
+import java.util.Iterator;
+import java.util.concurrent.CompletionStage;
+
+/**
+ *
+ * @since 13.0
+ **/
+public interface AsyncIterator<T> extends Iterator<CompletionStage<T>> {
```

Comment:
This will not work. The problem is you could have a command that returns an unkown amount of values. The iterator interface itself is blocking, so you would have to block waiting to confirm whether you have all the values or not. This really needs to be Flow.Publisher which provides a way to handle this with a callback and backpressure.

Reply:
Let's go with `Flow.Publisher`